src/buffer_manager.ts

```diff
@@ -622,8 +624,19 @@ export class BufferManager implements Disposable {
             this.textDocumentToBufferId.delete(document);
             unusedBuffers.push(bufId);
         });
-        unusedWindows.length && (await actions.lua("close_windows", unusedWindows));
-        unusedBuffers.length && (await actions.lua("delete_buffers", unusedBuffers));
+
+        // We don't await the result of closing these as a workaround for #2136,
+        // so that the user isn't blocked forever in case the nvim client gets stuck.
+        if (unusedWindows.length) {
+            actions
+                .lua("close_windows", unusedWindows)
+                .catch((err) => logger.warn(`Failed to cleanup windows ${inspect(unusedWindows)}: ${err}`));
```

Comment:
```suggestion
                .catch((err) => logger.warn("Failed to cleanup windows", unusedWindows, err));
```
The logger has already used inspect.